sway/main.c

```diff
@@ -303,25 +303,30 @@ int main(int argc, char **argv) {
 		}
 	}
 
+	// As the 'callback' function for wlr_log is equivalent to that for
+	// sway, we do not need to override it.
 	if (debug) {
-		wlr_log_init(WLR_DEBUG, NULL);
+		sway_log_init(SWAY_DEBUG, sway_terminate);
+		wlr_log_init(SWAY_DEBUG, NULL);
```

Comment:
This is wrong - wlr_log_init takes WLR_* arguments. You can also probably move this into sway_log_init and make the wlr log level implied by the sway log level (you'll need a mapping of sway loglevels to wlr loglevels)

Reply:
Thank you for catching that. As sway and wlroots are only used together by the main sway executable, not by swaymsg and others, having `sway_log_init` call `wlr_log_init` conditionally will be tricky to implement cleanly.

Reply:
Good point. Skip it, then.